src/ClientWrapper.php

```diff
@@ -50,41 +50,90 @@ public function getBranchClient(): BranchAwareClient
     }
 
     /**
-     * Returns branchClient if a branch was configured and basicClient otherwise.
+     * Returns branchClient if useBranchStorage flag is configured
      *
      * @return Client|BranchAwareClient
      */
-    public function getBranchClientIfAvailable(): Client
+    public function getTableAndFileStorageClient(): Client
     {
-        if ($this->hasBranch()) {
+        if ($this->clientOptions->useBranchStorage()) {
             return $this->getBranchClient();
         }
-
         return $this->getBasicClient();
     }
 
+    /**
+     * Returns branchClient if a branch was configured and basicClient otherwise.
+     *
+     * @return Client|BranchAwareClient
+     * @deprecated Use getBranchClient for all endpoints that support branches.
+     */
+    public function getBranchClientIfAvailable(): Client
+    {
+        return $this->getBranchClient();
```

Comment:
getBranchClientIfAvailable method is used where branches CAN be used, which means that it can always use branchClient (because branchId is always known now)